### PR TITLE
wallet-http: return 404 in getname

### LINKED ISSUE #356: wallet http name endpoints fail if name is unknown
Error message here could be cleaner:

```
$ curl x:api-key@127.0.0.1:14039/wallet/primary/name/fakenamethatdoesnotexist

{
  "error": {
    "message": "Cannot read property 'getJSON' of null"
  }
}
```

I think we can add some `if (!ns) {...}` checks in `wallet/http.js` in https://github.com/handshake-org/hsd/pull/163

ping @tynes

### PR DESCRIPTION
**Pull request by pinheadmz**
Closes #356

### CODE REVIEW

File: lib/wallet/http.js [hunk 1]
```diff
@@ -811,7 +811,10 @@ class HTTP extends Server {
       const network = this.network;
       const ns = await req.wallet.getNameStateByName(name);
 
-      res.json(200, ns.getJSON(height, network));
+      if (!ns)
+        return res.json(404);
```
Comment (boymanjor) [inline lib/wallet/http.js]: @pinheadmz the code base typically calls `res.json()` then calls `return`.

File: lib/wallet/http.js [hunk 2]
```diff
@@ -811,7 +811,10 @@ class HTTP extends Server {
...
+
+      return res.json(200, ns.getJSON(height, network));
```
Comment (boymanjor) [inline lib/wallet/http.js]: ditto

File: lib/wallet/http.js (hunk 1 above)
Comment (pinheadmz) [reply lib/wallet/http.js]: that's fine -- the error is `ns.getJSON` when `ns` is `null`

Comment (pinheadmz) [reply lib/wallet/http.js]: or do you mean you prefer an if/else instead of a return?

Comment (boymanjor) [reply lib/wallet/http.js]:
```js
if (!ns) {
  res.json(404);
  return;
}
```

Comment (pinheadmz) [reply lib/wallet/http.js]: hm, I think this is the convention though:

https://github.com/pinheadmz/hsd/blob/2ef54a503db4494270a190eed8910d8bc531d8ca/lib/wallet/http.js#L1142

although, I think I should change the error code to `400` not `404`

Comment (boymanjor) [reply lib/wallet/http.js]: I guess there isn't a convention lol: https://github.com/handshake-org/hsd/blob/2ef54a503db4494270a190eed8910d8bc531d8ca/lib/wallet/http.js#L218

Comment (boymanjor) [reply lib/wallet/http.js]: > although, I think I should change the error code to 400 not 404

Agreed.

Comment (pinheadmz) [reply lib/wallet/http.js]: ha! ok how about I just change the error code to `400` and we 🍻

Comment (boymanjor) [reply lib/wallet/http.js]: Discussed out of band. We're going to leave this as-is for now.

**Pull request merged**